prisma/seed.ts

```diff
@@ -0,0 +1,89 @@
+import { db } from '~/modules/database/db.server';
+
+const events = [
+  {
+    name: 'Tech Talks: Exploring the Latest Web Development Trends',
+    date: new Date('2023-07-19T08:00:00'),
+    description:
+      'Join us for an evening of insightful talks on progressive web apps, serverless architecture, and the future of JavaScript frameworks.',
+    imgUrl: 'imgs/upc-events1.png',
+    imgAlt: 'group of people in a meeting-1',
+    location: '123 Main St, San Francisco',
+    groupId: '345',
+  },
+  {
+    name: 'Fitness Bootcamp: Get Fit and Stay Active',
+    date: new Date('2023-08-15T18:30:00'),
+    description:
+      'Break a sweat and achieve your fitness goals with our high-intensity workout session suitable for all fitness levels.',
+    imgUrl: 'imgs/upc-events2.png',
+    imgAlt: 'group of people in a meeting-2',
+    location: '456 Elm St, Munich',
+    groupId: '346',
+  },
+  {
+    name: 'Tennis Tournament: Fun and Competitive Matches',
+    date: new Date('2023-09-22T12:15:00'),
+    description:
+      'Participate in our annual tennis tournament and showcase your skills in exciting matches with players from the community.',
+    imgUrl: 'imgs/upc-events3.png',
+    imgAlt: 'group of people in a meeting-3',
+    location: '789 Oak St, Istanbul',
+    groupId: '347',
+  },
+];
+
+const groups = [
+  {
+    id: '345',
+    name: 'WebDev Enthusiasts',
+    description:
+      'Join our community of web development enthusiasts and stay up-to-date with the latest trends, technologies, and innovations in the ever-evolving world of web development.',
+  },
+  {
+    id: '346',
+    name: 'FitFam Community',
+    description:
+      "Welcome to the FitFam Community! We're dedicated to helping you achieve your fitness goals and live a healthier, more active lifestyle. Join us for high-energy bootcamp sessions that are suitable for all fitness levels.",
+  },
+  {
+    id: '347',
+    name: 'Competitive Tennis Club',
+    description:
+      'Calling all tennis enthusiasts! Our Competitive Tennis Club is your gateway to exciting and friendly matches with players who share your passion for the game. Join us for fun and challenging competitive tennis tournaments.',
+  },
+];
+
+async function seed() {
+  try {
+    // Transform the events array into the format expected by createMany
+    const eventRecords = events.map((event) => ({
+      name: event.name,
+      date: event.date,
+      description: event.description,
+      imgUrl: event.imgUrl,
+      imgAlt: event.imgAlt,
+      location: event.location,
+      groupId: event.groupId,
+    }));
+
+    // Transform the groups array into the format expected by createMany
+    const groupRecords = groups.map((group) => ({
+      id: group.id,
+      name: group.name,
+      description: group.description,
+    }));
+
+    // Use createMany to insert records
+    await db.event.createMany({ data: eventRecords });
+    await db.group.createMany({ data: groupRecords });
```

Comment:
I would create the groups first as the events depend on the groups, no? It seems to work, but just as a good practice, I would change the order!

Reply:
I would suggest we also use the created groups with generated ids to then insert the events with the correct (uuid) group identifiers:

```javascript
const groups = await db.group.createMany({ data: groupRecords });
const eventRecords = events.map((event, index) => ({
   ...event,
   groupId: groups[index].id,
}));
await db.event.createMany({ data: eventRecords });
```

Reply:
Great point! Will fix it.